src/router.tsx

```diff
@@ -1,15 +1,23 @@
-import React from 'react';
+import React, { ReactElement } from 'react';
 import { BrowserRouter, Route, Routes } from 'react-router-dom';
 
+import RequireAuth from './components/auth/require-auth';
 import { APP_ROUTES } from './enums';
 import HomeLazy from './pages/home/home-lazy';
 import SigninLazy from './pages/sign-in/sign-in-lazy';
 
-const Router: React.FC = () => (
+const Router = (): ReactElement => (
   <BrowserRouter>
     <Routes>
+      <Route
+        path={APP_ROUTES.HOME}
+        element={
+          <RequireAuth>
+            <HomeLazy />
+          </RequireAuth>
+        }
+      />
```

Comment:
again - this logic can be done inside separate component `AuthorizedRoute`. Inside this component you will render children and check wheather it is authorized.
E.g.
```
<AuthorizedRoute path={APP_ROUTES.HOME}>
   <HomeLazy />
</AuthorizedRoute>
```

AuthorizedRoute file:
```
const AuthorizedRoute = ({children,...props}: RouteProps) => (
      <Route
        {...props}
        element={
          <RequireAuth>
            {children}
          </RequireAuth>
        }
      />
)
```

Reply:
I've used solution for it from here https://stackoverflow.com/questions/69864165/error-privateroute-is-not-a-route-component-all-component-children-of-rou